src/lib/components/controls/tabs/MappingsTab.svelte

```diff
@@ -80,29 +89,31 @@
 				uuid: data.uuid || editingMapping.uuid, // Use form UUID or fallback to original
 				justification: data.justification,
 				status: data.status,
-				source_entries: data.source_entries
+				source_entries: data.source_entries,
 			};
 
-			const uuidChanged = updatedMapping.uuid !== editingMapping.uuid;
-			
-			if (uuidChanged) {
-				const compositeId = `${control.id}:${editingMapping.uuid}`;
-				await wsClient.deleteMapping(compositeId);
-				await wsClient.createMapping(updatedMapping);
-			} else {
-				await wsClient.updateMapping(updatedMapping);
-			}
+			// hashes change every time so we just delete an create
+			await wsClient.deleteMapping(editingMapping.hash!);
+			delete updatedMapping.hash;
+			await wsClient.createMapping(updatedMapping);
 
 			resetMappingForm();
 		} catch (error) {
 			console.error('Failed to update mapping:', error);
 		}
 	}
 
-	async function handleDeleteMapping(uuid: string) {
+	async function handleDeleteMapping(hash: string) {
+		// Find the mapping by hash to get the UUID for backend deletion
+		const mappingToDelete = mappings.find(m => (m.hash === hash));
+		if (!mappingToDelete) {
+			console.error('Mapping not found for deletion');
+			return;
+		}
+
 		try {
-			const compositeId = `${control.id}:${uuid}`;
-			await wsClient.deleteMapping(compositeId);
+			// Backend expects UUID for file operations
+			await wsClient.deleteMapping(mappingToDelete.hash!);
```

Comment:
This is where I think if we passed the (new) composite key, we could skip looping over the cache on the backend.